internal/testdata/sampleHistory.json

```diff
@@ -0,0 +1,46 @@
+[
+  {
+    "eventId": 1,
+    "eventType": "WorkflowExecutionStarted",
+    "workflowExecutionStartedEventAttributes": {
+      "workflowType": {
+        "name": "go.uber.org/cadence/internal.testReplayWorkflow"
+      },
+      "taskList": {
+        "name": "taskList1"
+      }
+    }
+  },
+  {
+    "eventId": 2,
+    "eventType": "DecisionTaskScheduled",
+    "decisionTaskScheduledEventAttributes": {}
+  },
+  {
+    "eventId": 3,
+    "eventType": "DecisionTaskStarted"
+  },
+  {
+    "eventId": 4,
+    "eventType": "DecisionTaskCompleted",
+    "decisionTaskCompletedEventAttributes": {}
+  },
+  {
+    "eventId": 5,
+    "eventType": "ActivityTaskScheduled",
+    "activityTaskScheduledEventAttributes": {
+      "activityId": "0",
+      "activityType": {
+        "name": "testActivity"
+      },
+      "taskList": {
+        "name": "taskList1"
+      }
```

Comment:
Did you test to see if `input` and other binary fields are properly being round tripped?

Reply:
changed it to a sample history that has both input and timeouts.